ONSdigital/dp-frontend-filter-flex-dataset
# Feature/5683 populate collapsible on filter summary

### What

- Populated collapsible UI with data from dataset client 
- Refactored `handlers` tests into separate test files (lift and shift)
- Update api clients which fixes the `FilterOutputId` field not populating

### How to review

- Sense check
- Tests pass
- Review image and `data.json`
- _Optionally_ run the stack locally; using the [cantabular-import](https://github.com/ONSdigital/dp-compose/tree/main/cantabular-import) docker container. 
  - Create a dataset with metadata in the dimension description field(s) in Florence
  - Go to a [dataset landing page](http://localhost:8081/datasets/cantabular-flexible-example/editions/2021/versions/1)
  - Click 'change' 
  - Navigate 'back' to the summary page; should be something like `http://localhost:8081/filters/c039f4c7-a486-4511-85c8-0796e6743563/dimensions` 
  - Check the collapsible has real data 
  - **OR** call me 🤙 and I'll show you 

#### Image
<img width="394" alt="populated collapsible" src="https://user-images.githubusercontent.com/19624419/169013801-c3989d09-771f-49c5-9091-c17540e8fb27.png">

#### Data
```json
{
    "count": 2,
    "items": [
        {
            "description": "The age",
            "label": "Age",
            "links": {
                "code_list": {
                    "href": "http://api.localhost:23200/v1/code-lists/Age",
                    "id": "Age"
                },
                "options": {
                    "href": "http://api.localhost:23200/v1/datasets/Testing-server-v2/editions/2021/versions/1/dimensions/age/options",
                    "id": "age"
                },
                "version": {
                    "href": "http://api.localhost:23200/v1/datasets/Testing-server-v2/editions/2021/versions/1"
                }
            },
            "name": "age"
        },
        {
            "description": "The region",
            "label": "Region",
            "links": {
                "code_list": {
                    "href": "http://api.localhost:23200/v1/code-lists/Region",
                    "id": "Region"
                },
                "options": {
                    "href": "http://api.localhost:23200/v1/datasets/Testing-server-v2/editions/2021/versions/1/dimensions/geography/options",
                    "id": "geography"
                },
                "version": {
                    "href": "http://api.localhost:23200/v1/datasets/Testing-server-v2/editions/2021/versions/1"
                }
            },
            "name": "geography"
        }
    ],
    "limit": 20,
    "offset": 0,
    "total_count": 2
}
```

### Who can review

Frontend go dev
